File: python/rosalind_revc.py
def read_data(filepath: str) -> str:
    with open(filepath, mode='r') as file:
        return file.read().rstrip('\n')

def write_data(filepath: str, data: str) -> None:
    with open(filepath, mode='w') as file:
        file.write(data)

def get_reverse_complement(dna: str) -> str:
    complements = {'A':'T', 'T':'A', 'C':'G', 'G':'C'}
    return ''.join(complements[symbol] for symbol in reversed(dna))
    
if __name__ == '__main__':
    data = read_data('input/rosalind_revc.txt')
    result = get_reverse_complement(data)
    write_data('output/rosalind_revc.txt', result)

File: java/RosalindREVC.java
import java.io.IOException;
import java.nio.file.*;
import java.util.HashMap;

public class RosalindREVC {
    private static String readData(String filePath) throws IOException {
        Path inputFile = Paths.get(filePath);
        return Files.readAllLines(inputFile).get(0);
    }

    private static void writeData(String filePath, String data) throws IOException {
        Path outputFile = Paths.get(filePath);

        Files.deleteIfExists(outputFile);
        Files.createDirectories(outputFile.getParent());
        Files.createFile(outputFile);

        Files.writeString(outputFile, data);
    }

    private static String getReverseComplement(String dna) {
        HashMap<Character, Character> complements = new HashMap<>();
        complements.put('A', 'T');
        complements.put('T', 'A');
        complements.put('C', 'G');
        complements.put('G', 'C');

        StringBuilder reverseComplement = new StringBuilder(dna.length());

        for (int i = dna.length() - 1; i >= 0; i--) {
            reverseComplement.append(complements.get(dna.charAt(i)));
        }

        return reverseComplement.toString();
    }

    public static void main(String[] args) throws IOException {
        String data = readData("input/rosalind_revc.txt");
        String result = getReverseComplement(data);
        writeData("output/rosalind_revc.txt", result);
    }
}
